src/general.tsx

```diff
@@ -73,15 +74,45 @@ if (!useAppStore.getState().countrySet) {
             if (!requests) return;
             Object.entries(requests).forEach(async ([db_id, request]) => {
                 if (request.response_type) {
-                    useProceedingsStore.getState().addMessage({
+                    const messageFromRequest: Message = {
                         reference: request.reference,
                         recipient: request.recipient,
                         transport_medium: request.via,
                         type: request.response_type,
                         date: new Date(request.date),
-                        slug: request.slug,
                         email: request.email,
-                    });
+                    };
+
+                    const createStubProceeding = () => {
+                        // If the parent request is missing we create a stub to put the message in.
+                        useProceedingsStore.getState().addProceeding({
+                            reference: request.reference,
+                            messages: {},
+                            status: 'waitingForResponse',
+                        });
+                        useProceedingsStore.getState().addMessage(messageFromRequest);
+                    };
+
+                    // Search for the parent request in the already created proceedings
+                    if (Object.keys(useProceedingsStore.getState().proceedings).includes(request.reference))
+                        useProceedingsStore.getState().addMessage(messageFromRequest);
+                    else {
+                        // If no proceeding was found, search the database and create a new proceeding
+                        const parentRequestId = await (
+                            await userRequests.localforage_instance?.keys()
+                        )?.find((key) => key !== db_id && key.includes(request.reference));
+                        if (parentRequestId) {
+                            const parentRequest = await userRequests.getRequest(parentRequestId);
+                            if (!parentRequest) {
+                                createStubProceeding();
+                            } else {
+                                useProceedingsStore.getState().addProceeding(proceedingFromRequest(parentRequest));
+                                useProceedingsStore.getState().addMessage(messageFromRequest);
+
+                                await userRequests.removeRequest(parentRequestId);
+                            }
+                        } else createStubProceeding();
+                    }
```

Comment:
This will skip deleting the current entry. I guess not really necessary, though, since we get rid of the whole store later anyway. I just figured, if something fails during the migration and the store is not deleted, I didn't want to duplicate the data.

Reply:
I may be missing something but I really don't see a case where the two code snippets don't have the exact same behaviour.

Reply:
[This](https://github.com/datenanfragen/website/pull/911/files#diff-aa3b222de2ae407123deee1b1e641f51b54d8f655f9c463ed2e9337e864f3061R120) won't be run, if you return early.